subsys/bluetooth/shell/tbs.c

```diff
@@ -402,30 +504,43 @@ static int cmd_tbs_set_bearer_technology(const struct shell *sh, size_t argc,
 static int cmd_tbs_set_bearer_signal_strength(const struct shell *sh,
 					      size_t argc, char *argv[])
 {
-	long signal_strength;
-	long service_index;
-	int result;
+	unsigned long signal_strength;
+	unsigned long service_index;
+	int result = 0;
 
 	if (argc > 2) {
-		if (!strcmp(argv[1], "gtbs")) {
+		if (strcmp(argv[1], "gtbs") == 0) {
 			service_index = BT_TBS_GTBS_INDEX;
 		} else {
-			service_index = strtol(argv[1], NULL, 0);
-			if (service_index < 0 ||
-			service_index >= CONFIG_BT_TBS_BEARER_COUNT) {
-				shell_error(sh, "Invalid service index: %ld",
+			service_index = shell_strtol(argv[1], 0, &result);
+			if (result != 0) {
+				shell_error(sh,
+					    "Could not parse service_index: %d",
+					    result);
+
+				return -ENOEXEC;
+			}
+
+			if (service_index > CONFIG_BT_TBS_BEARER_COUNT) {
+				shell_error(sh, "Invalid service_index: %lu",
 					    service_index);
 
 				return -ENOEXEC;
 			}
 		}
 	} else {
-		service_index = 0;
+		service_index = 0U;
```

Comment:
Did you consider to factor out this repeated code to some helper function?

Reply:
I actually did :) It would make sense, but since this is a bugfixing PR, I'd rather keep the changes to a minimum. I suggest to do this in a separate PR.